packages/puppeteer-core/src/bidi/Page.ts

```diff
@@ -344,19 +344,19 @@ export class BidiPage extends Page {
     return await this.#cdpEmulationManager.emulateVisionDeficiency(type);
   }
 
-  override async setViewport(viewport: Viewport): Promise<void> {
+  override async setViewport(viewport: Viewport | null): Promise<void> {
     if (!this.browser().cdpSupported) {
       await this.#frame.browsingContext.setViewport({
-        viewport:
-          viewport.width && viewport.height
-            ? {
-                width: viewport.width,
-                height: viewport.height,
-              }
-            : null,
-        devicePixelRatio: viewport.deviceScaleFactor
-          ? viewport.deviceScaleFactor
+        viewport: viewport
```

Comment:
this should still check width and height since setting width and height to 0 resets the viewport

Reply:
I will fix it